### DISCUSSION TITLE
offline petsc build fails with ptscotch

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#24917] [Q&A Getting Started] [answered]
I'm using the offline build instructions on an offline HPC cluster. I'm getting an error when it's setting up ptscotch (undefined references to something with ptscotch in the name) during the petsc build when I use newer compiler/mpi/etc but it works fine with older tools. I know that's not a lot to go on, but since the cluster is air-gapped, the software is upgraded in waves. I can start digging if someone can suggest where to start.

**Comment by lindsayad** [comment]
@milljm have you run into this?

**Reply by jessecarterMOOSE** [reply]
Just got back from a trip and wanted to follow up. Looking closer at the error messages, looks like compiling ptscotch goes ok, but fails when linking. There are a bunch of undefined references to stuff with "scotch" in the name

**Reply by GiudGiud** [reply, marked as answer]
do you use ptscotch?
I think this may be one we can skip by modifying the script to ` --download-ptscotch=0`

**Reply by jessecarterMOOSE** [reply]
FWIW, I can build PETSc fine when I don't include ptscotch (set `--download-ptscotch=0` in scripts/configure_petsc.sh). I'm not even sure what this package does so not sure if I can get away with this in the long run.

**Reply by GiudGiud** [reply]
We can use ptscotch for partitioning. But it s not the default. I dont use it

Long run you certainly can live without it

**Reply by jessecarterMOOSE** [reply]
gotcha. Odd that is works with the remaining PETSc components

**Comment by jessecarterMOOSE** [comment]
@GiudGiud this is no longer an issue with latest petsc
